lib/OpenQA/WebAPI/Controller/Test.pm

```diff
@@ -259,59 +309,21 @@ sub _show {
     my ($self, $job) = @_;
     return $self->reply->not_found unless $job;
 
-    my $test_modules    = read_test_modules($job);
-    my $worker          = $job->worker;
-    my $clone_of        = $self->schema->resultset('Jobs')->find({clone_id => $job->id});
-    my $websocket_proxy = determine_web_ui_web_socket_url($job->id);
-
     $self->stash(
         {
-            job                     => $job,
-            testname                => $job->name,
-            distri                  => $job->DISTRI,
-            version                 => $job->VERSION,
-            build                   => $job->BUILD,
-            scenario                => $job->scenario_name,
-            worker                  => $worker,
-            assigned_worker         => $job->assigned_worker,
-            show_dependencies       => !defined($job->clone_id) && $job->has_dependencies,
-            clone_of                => $clone_of,
-            modlist                 => ($test_modules ? $test_modules->{modules} : []),
-            ws_url                  => $websocket_proxy,
-            has_parser_text_results => $test_modules->{has_parser_text_results},
+            job               => $job,
+            testname          => $job->name,
+            distri            => $job->DISTRI,
+            version           => $job->VERSION,
+            build             => $job->BUILD,
+            scenario          => $job->scenario_name,
+            worker            => $job->worker,
+            assigned_worker   => $job->assigned_worker,
+            clone_of          => $self->schema->resultset('Jobs')->find({clone_id => $job->id}),
+            show_dependencies => !defined($job->clone_id) && $job->has_dependencies,
+            show_autoinst_log => $job->state eq DONE && !$job->has_modules && $job->has_autoinst_log,
```

Comment:
Does that mean if a job has no modules,  do not show the autoinst log?

Reply:
No, if a job has no modules, *show* the log. This is basically the feature we introduced some months ago

Reply:
understand. Thanks.